testing/tests/test_context_on.py

```diff
@@ -115,9 +117,9 @@ def test_simple_secret_events(as_kwarg, event_name, event_kind, owner):
         mgr.run()
         assert len(mgr.charm.observed) == 2
         assert isinstance(mgr.charm.observed[1], ops.CollectStatusEvent)
-        mgr = mgr.charm.observed[0]
-        assert isinstance(mgr, event_kind)
-        assert mgr.secret.id == secret.id
+        evt = mgr.charm.observed[0]
+        assert isinstance(evt, event_kind)
+        assert evt.secret.id == secret.id
```

Comment:
Cheers!
I think I went a tad further here: https://github.com/canonical/operator/pull/1527/files#diff-104ae483d187a2a70cb0600c23ab87c65b3604059e64c0b695c51c77c128af06 though an improvement is still an improvement.

Reply:
Ah, I hadn't noticed that when I was looking over the tracing PR. Those changes are not actually related to the tracing, right? What about you/I cut those out of that PR and put them in a separate one, and then it reduces the size of the tracing one, and I can rebase/merge here and avoid the drive-by in this one too.

Reply:
Always a fan of reducing big PRs, thanks!

Reply:
#1596 for deferred events

Reply:
#1597 for this file

Reply:
Thanks for that! I'll pull those into this PR once they're merged.